subsys/pm/Kconfig

```diff
@@ -82,6 +82,24 @@ config PM_DEVICE_POWER_DOMAIN
 	  devices that depend on a domain will be notified when this
 	  domain is suspended or resumed.
 
+config PM_DEVICE_POWER_DOMAIN_DYNAMIC_NUM
+	int "Number of devices that can dynamically be bind to a power domain"
+	depends on PM_DEVICE_POWER_DOMAIN
+	default 0
+	help
+	  This option allows dynamically bind up to the given value
+	  number of devices to a Power Domain.
+
+# Since it is not possible to select HAS_DYNAMIC_DEVICE_HANDLES
+# directly from PM_DEVICE_POWER_DOMAIN_DYNAMIC_NUM, we have
+# this hidden symbol just to select HAS_DYNAMIC_DEVICE_HANDLES if
+# the number of power domain dynamic devices is greater than 0
+config HAS_PM_DEVICE_POWER_DOMAIN_DYNAMIC
+	bool
+	depends on PM_DEVICE_POWER_DOMAIN_DYNAMIC_NUM > 0
+	default y
+	select HAS_DYNAMIC_DEVICE_HANDLES
```

Comment:
I think it would be better to do something like this:

```
config PM_DEVICE_POWER_DOMAIN_DYNAMIC
	bool "Allow to dynamically bind devices to a power domain"
	depends on PM_DEVICE_POWER_DOMAIN
	select HAS_DYNAMIC_DEVICE_HANDLES
	help
	  ...

config PM_DEVICE_POWER_DOMAIN_DYNAMIC_NUM
	int "Number of devices that can dynamically be bind to a power domain"
	depends on PM_DEVICE_POWER_DOMAIN_DYNAMIC
	default 0 /* maybe can be set to 1 at least with suggested approach approach, as PM_DEVICE_POWER_DOMAIN_DYNAMIC_NUM is no longer used for boolean checks  */
	help
	  This option allows dynamically bind up to the given value
	  number of devices to a Power Domain.
```

Reply:
I think that is better indeed, more verbose since requires enable the feature and then select the number of slots, but it is clearer. Will change it.